packages/elements/src/react/utils/create-context-for-dom-validation.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+
+/**
+ * Use this context helper to detect whether a component has a particular parent higher up in the DOM or not.
+ */
+export function createContextForDomValidation(displayName: string) {
+  const ReactContext = React.createContext(false);
+  const OriginalProvider = ReactContext.Provider;
+
+  function Provider({ children }: WithChildrenProp) {
+    return React.createElement(
+      OriginalProvider,
+      {
+        value: true,
+      },
+      children,
+    );
+  }
+
+  Provider.displayName = displayName;
+
+  function useContext(allowMissingContext: boolean = false) {
+    const context = React.useContext(ReactContext);
+
+    if (!allowMissingContext && !context) {
+      throw new Error(
+        `You used a hook from "${Provider.displayName}" but it's not inside a <${Provider.displayName}.Provider> component.`,
+      );
```

Comment:
❓ Do we have a heuristic for when we throw a `ClerkElementsRuntimeError` vs just `Error`? 🤔

Reply:
I think at this point it's a bit random, but we can tackle it as part of SDK-1315